### PR TITLE
Cpp changes

### PR DESCRIPTION
**Pull request by ramensandwich**
Refactors Arduino code to combine common features into classes with appropriate hierarchy.  Also moves unused code from older systems into "unused" folder to prevent future confusion

### CODE REVIEW

File: Legacy/real_time/arduino_src/lib_avr/statuslights/StatusLights.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+/** @brief Turns off the light
+ *
+ */
+void StatusLights::Disable(void) {
+    *light_port &= ~_BV(register_offset);
+}
+
+void StatusLights::Flash(uint8_t time) {
+    //stubbed out
```
Comment (grypoB) [inline Legacy/real_time/arduino_src/lib_avr/statuslights/StatusLights.cpp]: do we want dead function in our code?

File: Legacy/real_time/arduino_src/radio_buggy_mega/Makefile [hunk 2]
```diff
@@ -109,7 +116,8 @@ EXTRAINCDIRS =
 #     gnu89 = c89 plus GCC extensions
 #     c99   = ISO C99 standard (not yet fully implemented)
 #     gnu99 = c99 plus GCC extensions
-CSTANDARD = -std=gnu99
+CSTANDARD = -std=iso9899:1999
+CPPSTANDARD = -std=c++11
```
Comment (grypoB) [inline Legacy/real_time/arduino_src/radio_buggy_mega/Makefile]: Which standards do we want to use? I switched the C one without real reason. Should we discuss this?

File: Legacy/real_time/arduino_src/radio_buggy_mega/Makefile [hunk 3]
```diff
@@ -467,51 +494,46 @@ extcoff: $(TARGET).elf
 
 
 # Link: create ELF output file from object files.
-.SECONDARY : $(TARGET).elf
```
Comment (grypoB) [inline Legacy/real_time/arduino_src/radio_buggy_mega/Makefile]: Even though it seems useless, should we apply the moto: don't fix what ain't broken?

File: Legacy/real_time/arduino_src/radio_buggy_mega/Makefile (hunk 2 above)
Comment (ramensandwich) [inline Legacy/real_time/arduino_src/radio_buggy_mega/Makefile]: If it ain't broke, don't fix it, IMO.

**Comment by ramensandwich** [comment]
This has been tested on the buggy by Sean, DTom, and Abha and seems to not have broken existing functionality.

**Pull request merged**